### PR TITLE
Ignore people arriving from localhost

### PR DESCRIPTION
**Pull request by gabebw**
Please check out #22 first.

When people click links on their site running on `http://localhost:3000`, we were adding `http://localhost` to the webring.

Now we don't add localhost requests to the webring, and we provide a helpful explanation of what's going on.

### CODE REVIEW

File: spec/controllers/redirections_controller_spec.rb [hunk 1]
```diff
@@ -28,6 +28,14 @@
         expect(first_redirection.reload.next).to eq new_redirection
       end
 
+      it "ignores requests from localhost" do
+        request.env["HTTP_REFERER"] = "http://localhost:3000"
+
+        get action, slug: "whatever"
```
Comment (adarsh) [inline spec/controllers/redirections_controller_spec.rb]: whatever

**Comment by adarsh** [comment]
:+1:

File: app/controllers/redirections_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class RedirectionsController < ApplicationController
+  before_action :ensure_referrer_is_not_localhost
```
Comment (edwardloveall) [inline app/controllers/redirections_controller.rb]: An alternative to the before action would be to:

* Make a validator that made sure the host wasn't localhost
* Raise a validation error if it was
* `rescue_from` that error in the app controller and redirect to localhost page

I've used this pattern in APIs to great effect, and mostly avoided before_actinos. Do you have a preference?

**Comment by gabebw** [comment]
Agh, closing #22 closed this and made it un-reopenable. GitHub!

**Comment by gabebw** [comment]
Reopened as #24.

**Pull request closed without merging**